dynamiq/memory/memory.py

```diff
@@ -0,0 +1,91 @@
+from datetime import datetime
+
+from dynamiq.memory.backend import InMemory, MemoryBackend
+from dynamiq.memory.config import Config
+from dynamiq.prompts import Message, MessageRole
+from dynamiq.utils.logger import logger
+
+
+class Memory:
+    """Manages the storage and retrieval of messages."""
+
+    def __init__(self, config: Config = Config(), backend: MemoryBackend = InMemory()):
+        """Initializes the Memory with the given configuration and backend.
+
+        If no backend is provided, an InMemory backend is used by default.
+        """
+        if not isinstance(backend, MemoryBackend):
+            raise TypeError("backend must be an instance of Backend")
+        self.config = config
+        self.backend = backend
+
+    def add(self, role: MessageRole, content: str, metadata: dict = None):
+        """Adds a message to the memory."""
+        try:
+            metadata = metadata or {}
+            metadata["timestamp"] = datetime.utcnow().timestamp()
+            message = Message(role=role, content=content, metadata=metadata)
+            self.backend.add(message)
+            logger.debug(
+                f"Memory {self.backend.name}: Added message: {message.role}: {message.content[:min(20, len(message.content))]}..."  # noqa: E501
```

Comment:
What do you think about adding an additional check for non-null content for this debug logging just in case? 
I remember I had similar case and issue later where it was okay for the message to be null (from a tool call or something similar), but my logger for debugging encountered an error because of that)